test/test_quant.py

```diff
@@ -0,0 +1,289 @@
+#    of its contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import torch
+from brevitas.core.quant import BinaryQuant, ClampedBinaryQuant, TernaryQuant
+from brevitas.core.quant import PrescaledRestrictIntQuantWithInputBitWidth, PrescaledRestrictIntQuant
+from brevitas.core.quant import RescalingIntQuant
+from brevitas.core.quant import IntQuant
+from brevitas.function.ops import min_int, max_int
+from brevitas.function.ops import tensor_clamp
+from brevitas.core.function_wrapper import TensorClamp, RoundSte, FloorSte, CeilSte
+import hypothesis.strategies as st
+from hypothesis import given
+from common import float_st, float_st_nz, two_lists_equal_size, list_float_st, float_st_p
+from common import ATOL, RTOL
+from unittest.mock import Mock
+from brevitas.core import ZERO_HW_SENTINEL_VALUE
+
+
+# EXECUTE WITH ENVIRONMENTAL VARIABLE PYTORCH_JIT=0
+
+MIN_BITWIDTH=2
+MAX_BITWIDTH=8
+
+# Used for BinaryQuant and ClampedBinaryQuant. The two tests are basically identical.
+def perform_test_binary(binary_type, inp, scaling):
+    scaling_impl_mock = Mock()
+    scaling_impl_mock.return_value = torch.tensor(scaling)
+
+    obj = binary_type(scaling_impl_mock)
+    output, _, _ = obj(inp, torch.tensor(ZERO_HW_SENTINEL_VALUE))
+    expected_output = torch.tensor([-1 * scaling, 1 * scaling])
+
+    # Check that output values match one of the possible values in expected_output
+    return check_admissible_values(output, expected_output) and check_binary_sign(inp, output)
+
+
+# Check that the output values are one of the admissible values. If not, return False and fail the test
+def check_admissible_values(predicted, admissible):
+    result = True
+    for value in predicted:
+        if value not in admissible:
+            result = False
+
+    return result
+
+
+# Check that after binarization the sign match the one expected given the input value
+def check_binary_sign(inp, predicted):
+    for value_input, value_predicted in zip(inp, predicted):
+        if ((value_input >= 0) and value_predicted < 0) or ((value_input < 0) and value_predicted > 0):
```

Comment:
can we rewrite it as not (value_input >= 0 and value_predicted > 0 or value input < 0 and value_predicted < 0)